src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2638,6 +2637,52 @@ theorem getElem_rotateLeft {x : BitVec w} {r i : Nat} (h : i < w) :
       if h' : i < r % w then x[(w - (r % w) + i)] else x[i - (r % w)] := by
   simp [← BitVec.getLsbD_eq_getElem, h]
 
+/-- If `w ≤ x < 2 * w`, then `x % w = x - w` -/
+private theorem add_mod_eq_add_sub {x w : Nat} (x_le :  w ≤ x) (x_lt : x < 2 * w) :
```

Comment:
Incorrectly named. Should be `mod_eq_sub`, or possibly `mod_eq_sub_of_le_of_lt` and made public.

(It's okay to use private theorems that belong in a different namespace, but preferrable to avoid them where possible.)